### ISSUE TITLE
Multiple axis reduction doesn't work

### ISSUE THREAD

**Issue by Hprairie** [Hprairie/tlib#1]
Let's take the basic example kernel:

```python
import torch
import triton
import triton.language as tl
import triton_lib as tlib
import triton_lib.functional as tlf

@triton.jit
def my_kernel(
    x_ptr,
    o_ptr,
    LENGHT: tl.constexpr,
):
    x = tl.load(x_ptr + tl.arange(0, LENGHT)[:, None] * LENGHT + tl.arange(0, LENGHT)[None, :])
    o = tlib.sum("[a b]", x)
    tl.store(o_ptr, x)


def launch(x):
    o = torch.zeros_like(x)
    print(x)
    my_kernel[(1, 1, 1)](x, o, x.shape[0])
    return o


x = torch.arange(8).to("cuda", dtype=torch.float32)
x = x[:, None] * 8 + x[None, :]
o = launch(x)
print(o)
```

Will throw an error. Need to either bake multiple axis reductions into the API or simply prevent it. Need to make sure that If I bake it in, that it doesn't lead to any poor optimizations.
